### ISSUE TITLE
`no-unused-imports` complains about decorator param

### ISSUE THREAD

**Issue by suevalov** [sweepline/eslint-plugin-unused-imports#92]
# Problem

`no-unused-imports` warns of a decorator parameter as an unused import.

Example of the code, where `of` is a violation of `no-unused-imports`

```typescript
@Resolver(of => Comment)
export class CommentResolver extends BaseResolver {
}
```

## Versions

Package versions:

```json
    "@typescript-eslint/eslint-plugin": "7.7.0",
    "@typescript-eslint/parser": "7.10.0",
    "eslint": "8.56.0",
    "eslint-plugin-unused-imports": "3.2.0",
```

# Expected

It should only be taken care by `unused-imports/no-unused-vars` rule.

**Comment by charlescrain** [comment]
Same issue with:

```
 "eslint": "9.8.0",
 "eslint-plugin-unused-imports": "4.0.1",
 "@typescript-eslint/eslint-plugin": "8.0.1",
 "@typescript-eslint/parser": "8.0.1",
 ```

**Comment by antfu** [comment]
We temporarily closed this due to the lack of enough information. We could not identify whether it was a bug or a userland misconfiguration with the given info.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)
